### PR TITLE
Create the ManageSitePluginsDialog and use it on the Manage Sites button

### LINKED ISSUE #63888: Plugin Details: Manage site plugins
### Details

Details
When in a multi-site context and viewing a plugin that has been installed on at least 1 site, I want to be able to control the status of that plugin across sites. f84fqtCG4pCnCh2Hq0But5-fi-1953%3A23616

This task moves the existing "Installed on" table into a modal triggered by a "manage sites" button within the sidebar.

image
image

Acceptance
- When viewing a plugin outside a single site (e.g. https://wordpress.com/plugins/woocommerce-subscriptions) and that plugin has been purchased and installed on at least 1 site, I see a "manage sites" button within the sidebar, as well as a "Installed and active on 3 sites" label
- Clicking manage sites button triggers a modal
- Inside the modal I can see the current table of all my sites and whether the plugin has been installed on those sites
- All existing actions work as previously

### Checklist

- [ ]
- [ ]

### Related

_No response_

### PR DESCRIPTION
**Pull request by WBerredo**
#### Proposed Changes

* Create the ManageSitePluginsDialog component with the Site List and its functions (activate, autoupdate, remove, settings, etc)
* Use it on the Plugin Details page

Screen Shot 2022-08-05 at 17 11 36

#### Testing Instructions

* Enable the feature flag `plugins/plugin-details-layout`*
* Go to a multi-site plugin page (`/plugins/{plugin_slug}`)
* Verify you see the `Manage sites` button
* Click on the button 
* You should see a section with all sites available (atomic), being the plugin installed or uninstalled.
* Test all the available options (activate, autoupdate, remove, settings, etc)

\* To enable this feature flag add `?flags=plugins/plugin-details-layout` to the end of your URL or past the following code on the developer tools console `document.cookie = 'flags=plugins/plugin-details-layout;max-age=1209600;path=/'`

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

---

Fixes #63888

### CODE REVIEW

File: client/my-sites/plugins/plugin-details-CTA/style.scss [hunk 1]
```diff
@@ -142,6 +142,46 @@
...
+	.plugin-action .components-toggle-control .components-base-control__field {
+		margin-bottom: 0;
+	}
+
+	.manage-site-plugins-dialog__plugin-list {
+		margin-bottom: 16px;
+	}
+
+	.manage-site-plugins-dialog__finish-button {
+		float: right;
```
Comment (cpapazoglou) [inline client/my-sites/plugins/plugin-details-CTA/style.scss]: Cam we move this to a separate css file related to the new component? Also, is this component bound to the plugin details CTA? Would it make sense moving it to a separate folder?

File: client/my-sites/plugins/plugin-details-CTA/style.scss [hunk 2]
```diff
@@ -159,3 +199,7 @@
 		@extend %placeholder;
 	}
 }
+
+.ellipsis-menu__menu.popover {
```
Comment (cpapazoglou) [inline client/my-sites/plugins/plugin-details-CTA/style.scss]: `z-index` is best declared in `client/assets/stylesheets/shared/functions/_z-index.scss`

File: client/my-sites/plugins/plugin-details-CTA/manage-site-plugins-dialog.jsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+import { siteObjectsToSiteIds } from 'calypso/my-sites/plugins/utils';
+import { getBillingInterval } from 'calypso/state/marketplace/billing-interval/selectors';
+import {
+	getSiteObjectsWithPlugin,
+	getSiteObjectsWithoutPlugin,
+} from 'calypso/state/plugins/installed/selectors';
+import getSelectedOrAllSitesWithPlugins from 'calypso/state/selectors/get-selected-or-all-sites-with-plugins';
+import siteHasFeature from 'calypso/state/selectors/site-has-feature';
+
+export function ManageSitePluginsDialog( { isVisible, onClose, plugin } ) {
```
Comment (cpapazoglou) [inline client/my-sites/plugins/plugin-details-CTA/manage-site-plugins-dialog.jsx]: Can we use an arrow function here to increase parity with the rest of the codebase?

**Review by cpapazoglou — CHANGES_REQUESTED** [review]
1. Can we add a little bit more padding to each row so that buttons don't get stacked?
CleanShot 2022-08-09 at 11 48 57@2x

2. Let's change the copy of the `Finished` button to `Close`
CleanShot 2022-08-09 at 11 51 32@2x

3. Responsive styles are missing. We should always make sure that interfaces look great in all devices
CleanShot 2022-08-09 at 12 01 21@2x

File: client/my-sites/plugins/plugin-details-CTA/style.scss (hunk 2 above)
Comment (WBerredo) [reply client/my-sites/plugins/plugin-details-CTA/style.scss]: Implemented at https://github.com/Automattic/wp-calypso/pull/66282/commits/baea81a75bd82b58401be5a46abba55555105248

File: client/my-sites/plugins/plugin-details-CTA/manage-site-plugins-dialog.jsx (hunk 3 above)
Comment (WBerredo) [reply client/my-sites/plugins/plugin-details-CTA/manage-site-plugins-dialog.jsx]: Done at https://github.com/Automattic/wp-calypso/pull/66282/commits/baea81a75bd82b58401be5a46abba55555105248

File: client/my-sites/plugins/plugin-details-CTA/style.scss (hunk 1 above)
Comment (WBerredo) [reply client/my-sites/plugins/plugin-details-CTA/style.scss]: Done at https://github.com/Automattic/wp-calypso/pull/66282/commits/baea81a75bd82b58401be5a46abba55555105248

**Comment by WBerredo** [comment]
The mobile layout was updated in a similar manner as we have today, I'm waiting for confirmation from @SaxonF if that works for now.

Screen Shot 2022-08-10 at 10 22 00

Screen Shot 2022-08-10 at 10 20 40

**Review by cpapazoglou — APPROVED** [review]
Thanks for resolving the feedback Wesley, it looks good now!

**Pull request merged**